src/mongo/mongoConnectionStrings.ts

```diff
@@ -48,18 +49,18 @@ export async function parseMongoConnectionString(connectionString: string): Prom
     let host: string;
     let port: string;
 
-    const db = await connectToMongoClient(connectionString, appendExtensionUserAgent());
-    const serverConfig = db.serverConfig;
+    const mongoClient: MongoClient = await connectToMongoClient(connectionString, appendExtensionUserAgent());
+    const serverConfig: Server | ReplSet | Mongos = mongoClient.db(testDb).serverConfig;
     // Azure CosmosDB comes back as a ReplSet
     if (serverConfig instanceof ReplSet) {
-        // get the first connection string from the seedlist for the ReplSet
+        // get the first connection string from the servers for the ReplSet
         // this may not be best solution, but the connection (below) gives
         // the replicaset host name, which is different than what is in the connection string
         // "s" is not part of ReplSet static definition but can't find any official documentation on it. Yet it is definitely there at runtime. Grandfathering in.
         // tslint:disable-next-line:no-any
         let rs: any = serverConfig;
-        host = rs.s.replset.s.seedlist[0].host;
-        port = rs.s.replset.s.seedlist[0].port;
+        host = rs.s.options.servers[0].host;
+        port = rs.s.options.servers[0].port;
     } else {
         host = serverConfig['host'];
```

Comment:
So the top case changed - I wonder if the else case also changed? That would be bad if we only work in the Azure case and break other mongo dbs

Reply:
Do you just test this with a local mongodb instance running?

Reply:
I haven't tested Cosmos in a while haha... but that sounds like a good option

Reply:
![image](https://user-images.githubusercontent.com/5290572/65988192-3e8b5480-e43c-11e9-864d-c5f9836ca432.png)

![image](https://user-images.githubusercontent.com/5290572/65988251-5367e800-e43c-11e9-8277-8700cf647b60.png)

We good.